### PR TITLE
Fix #934: typing of agregation functions when a machine word isn't 64-bit

### LINKED ISSUE #934: Test failure on 32-bit Linux
```
======================================================================
FAIL: test_prod_magnitude (__main__.TestArrayMethods)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/antoine/numba/numba/tests/support.py", line 140, in assertPreciseEqual
    self._assertPreciseEqual(first, second, prec, ulps, msg)
  File "/home/antoine/numba/numba/tests/support.py", line 228, in _assertPreciseEqual
    _assertNumberEqual(first, second, delta)
  File "/home/antoine/numba/numba/tests/support.py", line 158, in _assertNumberEqual
    self.assertEqual(first, second, msg=msg)
AssertionError: -1373712384 != -4696715432980197376

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/antoine/numba/numba/tests/test_array_methods.py", line 306, in test_prod_magnitude
    self.check_aggregation_magnitude(array_prod)
  File "/home/antoine/numba/numba/tests/test_array_methods.py", line 295, in check_aggregation_magnitude
    self.assertPreciseEqual(npr, nbr)
  File "/home/antoine/numba/numba/tests/support.py", line 145, in assertPreciseEqual
    self.fail("when comparing %s and %s: %s" % (first, second, failure_msg))
AssertionError: when comparing -1373712384 and -4696715432980197376: -1373712384 != -4696715432980197376
```

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
There's still a failure on win-64...

**Comment by pitrou** [comment]
The win-64 failure is due to Numba producing better code than Numpy. Under win-64, Numpy's prod() uses 32-bit computations, while Numba uses 64-bit ones.

**Comment by seibert** [comment]
We should open up a bug report in their issue tracker...  :)

File: numba/tests/test_array_methods.py [hunk 1]
```diff
@@ -284,17 +284,22 @@ def test_array_prod_global(self):
...
-    def check_aggregation_magnitude(self, pyfunc):
+    def check_aggregation_magnitude(self, pyfunc, is_prod=False):
         """
         Check that integer overflows are avoided (issue #931).
         """
-        # Overflows are always avoided here (ints are cast either to int64
+        # Overflows are avoided here (ints are cast either to intp
         # or float64).
-        arr = np.arange(10, dtype='int16') + 60000
+        n_items = 2 if is_prod else 10
```
Comment (seibert) [inline numba/tests/test_array_methods.py]: Can you add a comment here explaining why this workaround is needed?

**Comment by pitrou** [comment]
I've reported the Numpy issue at https://github.com/numpy/numpy/issues/5448

**Comment by pitrou** [comment]
And I added a comment to explain the workaround in the tests.

**Pull request merged**